nexus/src/main/java/com/example/nexus/controller/AdminController.java

```diff
@@ -102,4 +103,11 @@ public ResponseEntity<?> addCampaign(@RequestBody CampaignRequest campaignReques
 
         return ResponseEntity.status(HttpStatus.CREATED).build();
     }
+
+    @PatchMapping("/product/{productId}")
+    public ResponseEntity<?> editProduct(@PathVariable Long productId, @Valid @ModelAttribute ProductRequest productRequest) {
+        this.EditProductService.editProduct(productId, productRequest);
+
+        return ResponseEntity.status(HttpStatus.OK).build();
+    }
 }
```

Comment:
Add one more patch for change product campaignDsicount. If campaign is active change disccount too.